### PR TITLE
[improvements] Reformat strings to use f-strings #107

### LINKED ISSUE #107: [improvements] Use f-string for all strings
Check comment here: https://github.com/openwisp/docker-openwisp/pull/101#discussion_r382830175

### PR DESCRIPTION
**Pull request by Purhan**
Closes #107

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -15,12 +15,12 @@ def change_secret_key(keygen):
     file_string = file_handle.read()
     file_handle.close()
     file_string = re.sub(
-        r'DJANGO_SECRET_KEY=.*', r'DJANGO_SECRET_KEY=' + keygen, file_string
+        r'DJANGO_SECRET_KEY=.*', fr'DJANGO_SECRET_KEY={keygen}', file_string
```
Comment (atb00ker) [inline build.py:18]: Quick question, have you tested this line? :smile:

File: build.py [hunk 2]
```diff
@@ -15,12 +15,12 @@ def change_secret_key(keygen):
...
     )
     if file_string[-1] != '\n':
         file_string += '\n'
     if 'DJANGO_SECRET_KEY' not in file_string:
-        file_string += 'DJANGO_SECRET_KEY=' + keygen
+        file_string += 'DJANGO_SECRET_KEY={}'.format(keygen)
```
Comment (atb00ker) [inline build.py:23]: Why is `format` used, can we use f-string here?
eg:
`f'DJANGO_SECRET_KEY={keygen}'`

File: build/common/openwisp/settings.py [hunk 3]
```diff
@@ -86,7 +86,7 @@
 ASGI_APPLICATION = 'openwisp_controller.geo.channels.routing.channel_routing'
 
 REDIS_HOST = os.environ['REDIS_HOST']
-CELERY_BROKER_URL = 'redis://' + REDIS_HOST + ':6379/1'
+CELERY_BROKER_URL = 'redis://{}:6379/1'.format(REDIS_HOST)
```
Comment (atb00ker) [inline build/common/openwisp/settings.py:89]: Why is format used, can we use `f-string` here?

File: build/common/openwisp/settings.py [hunk 4]
```diff
@@ -128,7 +128,7 @@
 CACHES = {
     'default': {
         'BACKEND': 'django_redis.cache.RedisCache',
-        'LOCATION': 'redis://' + REDIS_HOST + ':6379/1',
+        'LOCATION': 'redis://{}:6379/1'.format(REDIS_HOST),
```
Comment (atb00ker) [inline build/common/openwisp/settings.py:131]: Why is format used, can we use `f-string` here?

**Review by atb00ker — CHANGES_REQUESTED** [review]
Okay, format is used in a lot of places, any particular reason for that.
Also, please double check if you have covered all the files. I was expecting more files, however, I might be wrong about this.


Also, finally your build is failing, check lines:

1. https://travis-ci.org/github/openwisp/docker-openwisp/builds/728988097#L424
2. https://travis-ci.org/github/openwisp/docker-openwisp/builds/728988097#L442

About the (2) error, 

```
 [improvements] Reformat strings to use f-strings #107

Closes #107  
```

should solve it. :-)

**Comment by Purhan** [comment]
>     1. Okay, format is used in a lot of places, any particular reason for that? We want to minimize the use of format, it's very verbose!

My bad, I misunderstood the issue. I have replaced all the strings with f-strings now.


>     2. Please double check if you have covered all the files. I was expecting more files, however, I might be wrong about this.

Running `grep --regexp="[\"\']\s*\+|.format\(" --extended-regexp --recursive --line-number docker-openwisp` does not return any files now. I don't know if there is a better way to check.

**Comment by Purhan** [comment]
>     1. Okay, format is used in a lot of places, any particular reason for that? We want to minimize the use of format, it's very verbose!

My bad, I misunderstood the issue. I have replaced all the strings with f-strings now.


>     2. Please double check if you have covered all the files. I was expecting more files, however, I might be wrong about this.
Running `grep --regexp="[\"\']\s*\+|.format\(" --extended-regexp --recursive --line-number docker-openwisp` does not return any files now. I don't know if there is a better way to check.

File: build.py (hunk 1 above)
Comment (Purhan) [reply build.py:18]: Below is the output of `make runtests`. Kindly tell me if there are other tests that I have to run.
Screenshot from 2020-09-22 13-04-42

**Comment by Purhan** [comment]
>     3. finally your build is failing, check lines:
>        3.1. https://travis-ci.org/github/openwisp/docker-openwisp/builds/728988097#L424
>        3.2. https://travis-ci.org/github/openwisp/docker-openwisp/builds/728988097#L442
> 
> 
> About the (3.2) error,
> 
> ```
> [improvements] Reformat strings to use f-strings #107
> 
> Closes #107  
> ```

Done! All the checks have successfully passed. Please take a look. 
Thanks. :smile:

**Comment by Purhan** [comment]
@atb00ker Just a reminder (you possibly didn't get notified), please take a look. Thanks again.

**Comment by atb00ker** [comment]
@Purhan I'll review this after I review the other PRs, feel free to pick up another issue in the meanwhile. :smile:

**Review by atb00ker — CHANGES_REQUESTED** [review]
@Purhan , in the meanwhile, I see that you have made 3 commits: https://github.com/openwisp/docker-openwisp/pull/128/commits

Please use `git rebase` to squash all your commits to one commit.

**Comment by atb00ker** [comment]
@Purhan why is this closed? Are you having problem in rebasing it ? :smile: 

Do you need help with this same?

**Comment by Purhan** [comment]
@atb00ker Yes, as you can see when I tried to rebase, I accidentally changed hashes for previous commits unrelated to this pull request. Anyways, I have opened a new PR #129 , please take a look at that one. Thanks.

**Comment by atb00ker** [comment]
Okay! :smile: 

However, learning git is important, so try to not close PRs, if you need help next time, feel free to ask! :smile:

**Pull request closed without merging**